include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,204 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/include/util.hpp>
+
+#include "dlaf/blas/tile.h"
+#include "dlaf/common/index2d.h"
+#include "dlaf/common/round_robin.h"
+#include "dlaf/common/vector.h"
+#include "dlaf/eigensolver/backtransformation/api.h"
+#include "dlaf/factorization/qr.h"
+#include "dlaf/matrix/copy.h"
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/matrix/layout_info.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/matrix/panel.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace eigensolver {
+namespace internal {
+
+template <Device device, class T>
+void copySingleTile(hpx::shared_future<matrix::Tile<const T, device>> in,
+                    hpx::future<matrix::Tile<T, device>> out) {
+  hpx::dataflow(dlaf::getCopyExecutor<Device::CPU, Device::CPU>(),
+                matrix::unwrapExtendTiles(matrix::copy_o), in, out);
+}
+
+template <class Executor, Device device, class T>
+void trmmPanel(Executor&& ex, hpx::shared_future<matrix::Tile<const T, device>> t,
+               hpx::future<matrix::Tile<T, device>> w) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::trmm_o), blas::Side::Right, blas::Uplo::Upper,
+                blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), t, w);

```

Comment:
What do you think @rasolca?
```suggestion
                blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), t, std::move(w));
```